docs/farming-&-staking/farming/advanced-cli/cli-farming-cluster.mdx

```diff
@@ -12,137 +13,420 @@ keywords:
+
+<details open>
+<summary>Controller (Manages node communication and orchestrates other components.)</summary>
+
+The first component to start is the Controller. It connects to a single node, and you must specify the node's location on the network.
+
+:::caution Proper Port Forwarding
+
+For your cluster to be reachable, open and forward the default port `30533` to the Controller. If you've changed the port or are using multiple controllers, set it with `--listen-on /ip4/0.0.0.0/tcp/<port>`.
+
+Refer to our guide on [Port Forwarding](/farming/guides/port-config).
+
+:::
+
+```bash
+./subspace-farmer \
+    cluster \
+        --nats-server nats://<nats_ip>:4222 \
     controller \
         --base-path /path/to/controller-dir \
-        --node-rpc-url ws://<NODE_IP>:<NODE_PORT>
+        --node-rpc-url ws://<node_ip>:<rpc_port>
 ```
 
-Replace **`<NATS_IP>`** with your NATS server IP address, and **`<NODE_IP>:<NODE_PORT>`** with your node’s IP address and port. Specify the working directory with **`--base-path`**.
+Replace the following placeholders:
+
+- `<nats_ip>`: The IP address of your NATS server.
+- `/path/to/controller-dir`: The path where the controller's peer network details will be saved.
+- `<node_ip>` and `<rpc_port>`: Your node's IP address and RPC port.
 
-The controller logs farm connections, disconnections, and piece cache sync progress. Optional connection-related options include **`--in-connections`**, **`--out-connections`**, **`--pending-in-connections`**, and **`--pending-out-connections`**.
+<details>
+<summary>Common Paramaters</summary>
 
-### Cache
+Refer to `./subspace-farmer cluster controller --help` for an explanation of each option and its default values.
 
-Next, run the cache component. Although you can run multiple cache processes to distribute the load, only one is required.
+```shell
+--listen-on /ip4/0.0.0.0/tcp/<port>
+--listen-on /ip6/::1/tcp/<port>
 
+--cache-group
+--in-connections
+--out-connections
+--pending-in-connections
+--pending-out-connections
 ```
-subspace-farmer cluster --nats-server nats://<NATS_IP>:4222 \
+</details>
+
+</details>
+
+<details open>
+<summary>Cache (Downloads sector data from DSN peers and serves it to the plotter.)</summary>
+
+:::caution Cache Size
+
+The Cache stores the blockchain's Archive History. For optimal performance, the cache must be large enough to hold this data locally. If the data isn't available in the cache, plotting may slow down as it has to be fetched from peers over the DSN. Currently, we recommend a minimum of 200GiB.
+
+The current Archive History Size can be found in the top cards on [Astral Block Explorer](https://astral.autonomys.xyz).
+
+:::
+
+```bash
+./subspace-farmer \
+    cluster \
+        --nats-server nats://<nats_ip>:4222 \
     cache \
-        path=/path/to/cache,size=SIZE
+        path=/path/to/cache,size=<size>
```

Comment:
This is obvious to us, but I think it'd be nice to explain that more than one of these can be specified and how just like we have it for farms.